File: api/trigger.py
from flask import Flask, jsonify, request #Force Deploy
import os
import requests

app = Flask(__name__)

GITHUB_TOKEN = os.getenv("GITHUB_TOKEN")
REPO = "BillyLikesBacon/mcserverstarter"
WORKFLOW_FILE = "main.yml"
BRANCH = "main"

@app.route("/", methods=["POST"])
def trigger():
    url = f"https://api.github.com/repos/{REPO}/actions/workflows/{WORKFLOW_FILE}/dispatches"
    headers = {
        "Authorization": f"Bearer {GITHUB_TOKEN}",
        "Accept": "application/vnd.github+json"
    }
    data = {"ref": BRANCH}

    response = requests.post(url, json=data, headers=headers)
    if response.status_code == 204:
        return jsonify({"status": "Workflow triggered!"})
    else:
        return jsonify({"status": "Failed", "detail": response.json()}), 400

def main(vercel_request):
    with app.test_request_context(
        path=vercel_request.path,
        base_url=vercel_request.base_url,
        query_string=vercel_request.query_string,
        method=vercel_request.method,
        headers=vercel_request.headers,
        data=vercel_request.get_data()
    ):
        return app.full_dispatch_request()

File: api/trigger.js
import fetch from "node-fetch";

export default async function handler(req, res) {
  if (req.method !== "POST") {
    res.status(405).send({ error: "Only POST allowed" });
    return;
  }

  const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
  const REPO = "BillyLikesBacon/mcserverstarter";
  const WORKFLOW_FILE = "main.yml";
  const BRANCH = "main";

  try {
    const response = await fetch(
      `https://api.github.com/repos/${REPO}/actions/workflows/${WORKFLOW_FILE}/dispatches`,
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${GITHUB_TOKEN}`,
          Accept: "application/vnd.github+json",
        },
        body: JSON.stringify({ ref: BRANCH }),
      }
    );

    if (response.status === 204) {
      // Success, no content returned
      res.status(200).json({ status: "Workflow triggered!" });
    } else {
      // Only try to parse JSON if there is content
      const text = await response.text();
      let data;
      try {
        data = JSON.parse(text);
      } catch {
        data = { raw: text };
      }
      res.status(response.status).json({ status: "Failed", detail: data });
    }
  } catch (err) {
    res.status(500).json({ status: "Error", detail: err.message });
  }
}